python/lsst/ts/FiberSpectrograph/fiberSpectrograph.py

```diff
@@ -141,3 +231,208 @@ def disconnect(self):
 
     def __del__(self):
         self.disconnect()
+
+    def _configure_ctypes(self):
+        """Configure function arguments for communication with libavs.
+
+        Some of the functions in libavs need to have their types and/or
+        return values explicitly defined as pointers for python to be able
+        to pass them in correctly.
+        """
+        self.libavs.AVS_Activate.argtypes = [ctypes.POINTER(AvsIdentity)]
+        self.libavs.AVS_GetParameter.argtypes = [ctypes.c_int,
+                                                 ctypes.c_uint,
+                                                 ctypes.POINTER(ctypes.c_uint),
+                                                 ctypes.POINTER(DeviceConfig)]
+        self.libavs.AVS_GetAnalogIn.argtypes = [ctypes.c_int,
+                                                ctypes.c_ubyte,
+                                                ctypes.POINTER(ctypes.c_float)]
+
+
+# size of these character fields in bytes
+AVS_USER_ID_LEN = 64
+AVS_SERIAL_LEN = 10
+
+
+class AvsIdentity(ctypes.Structure):
+    """Python structure to represent the `AvsIdentityType` C struct."""
+    size = 75  # total size of this structure in bytes
```

Comment:
I added `size` so that my class can use it to tell the C code the size of the incoming data structure (e.g. `required_size`). The AVS documentation says how big each of these structures is in bytes.

Reply:
So what about computing it? Perhaps a function that takes a ctypes Structure and emits the size (at which point it doesn't even have to be an attribute)? Does `ctypes.sizeof(struct)` work?

Reply:
Thanks for pestering me about this. It turns out that `ctypes.sizeof` does appear to do what I want.